lib/rust/ensogl/core/src/display/garbage.rs

```diff
@@ -0,0 +1,106 @@
+//! A module with garbage [`Collector`].
+use crate::prelude::*;
+
+
+
+// ===============
+// === Garbage ===
+// ===============
+
+/// The structure with collected garbage arranged in three buckets.
+///
+/// For more info, see the docs of [`Collector`].
+#[derive(Debug, Default)]
+struct Garbage {
+    before_pixel_sync:   Vec<Box<dyn Any>>,
+    before_pixel_update: Vec<Box<dyn Any>>,
+    before_mouse_events: Vec<Box<dyn Any>>,
+}
+
+
+/// The Garbage Collector
+///
+/// This structure collects the EnsoGL components structures designated to being drop once the
+/// component's hiding will be finished and all related events will be handled. Thanks to that,
+/// the library's user don't need to handle "is dropped" events separately from "is hidden" events.
+///
+/// # Implementation
+///
+/// Each collected component FRP network and model must be thus kept long enough until the following
+/// sequence will happen:
+/// 1. We render the scene without the component;
+/// 2. On the new rendered scene, we request the pixel value under the mouse cursor (see
+/// [`crate::display::render::passes::PixelReadPass`]).
+/// 3. The requested pixel value is received
+/// 4. The mouse events are handled (see [`crate::display::scene::SceneData::handle_mouse_events`]).
```

Comment:
1. mixed semicolons, dots, and nothing at the end of sentences.
2. Point 3 - it should be explained that "received" means that it is read into CPU memory after creating fences on GPU. Basically, this docs lacks information that the whole process can span over several frames and that later frames can get pixels from a few frames back, which is crucial here.